File: HW/HW2/isBad.java
package HW.HW2;

public class isBad {
    public int firstBadVersion(int n) {
        int left = 1;
        int right = n;
        while (left <= right)
        {
            int mid = (right - left)/2 + left;
            if (isBadVersion(mid) && !isBadVersion(mid-1))
            {
                return mid;
            }
            else if (isBadVersion(mid))
            {
                //Left
                right = mid;
            }
            else
            {
                //Right
                left = mid+1;
            }
        }
        return left;
    }
    
    boolean isBadVersion(int version){
        return true;
    }

}

File: HW/HW2/isBad.py
class isBad(object):
    #To whomever is grading, not sure how the helper isBadVersion is implemented so I simply inserted it into the algo
    def firstBadVersion(self, n: int):
        left = 1
        right = n
        def isBadVersion(self, n):
            return True
        while left < right:
            mid = (right - left)//2 + left
            if isBadVersion(mid) and not isBadVersion(mid-1):
                return mid
            elif isBadVersion(mid):
                right = mid
            else:
                left = mid + 1
        return left
        


   
